### PR TITLE
Set up zeit/swr and reducers

### PR DESCRIPTION
**Pull request by bdougie**
# What is this?

1. Adds zeit/swr to accomplish remote data fetching
2. introduces reducer pattern to pair with data fetching
3. No tests

# Why?
I started looking into how to reduce hits to the api.github.com/graphql. I was able to mostly fix by [updating the default values in the useEffect](https://github.com/bdougie/open-sauced/commit/ff27f584fbd8436acf39f0b5dc977d9ef3ad8298#diff-ec976aaee2548bdfb5fb02cc65e88f5dL26) and properly using hooks.

## failed apollo cache implementation
I attempted to add apollo-client caching, outlined in https://github.com/bdougie/open-sauced/issues/348#issuecomment-551196791. However, because I am using [onegraph-auth](https://github.com/OneGraph/onegraph-client/issues/18) my pattern for leveraging Apollo is conflicting with the basic implementation.  

## implement [zeit/swr](https://swr.now.sh/)
I found out about swr a few months back and decided to leverage in place of apollo in-memory caching. I basically needed a cache, but it also gives you a few free things to React. Mainly the easy Suspense option, remote fetching with hooks, and built-in caching. 

While adding SWR, I also added a reducer pattern  

## TODO (what is next?): 
- [Test Suspense](https://www.youtube.com/watch?v=lfb5jvHq9c4) #408 
- [Test reducers](https://medium.com/@netxm/testing-redux-reducers-with-jest-6653abbfe3e1) #409
- Implement swr for Issues.js fetching and pagination.

### CODE REVIEW

**Comment by NileshPatel17** [comment]
was checking this build with link ```https://deploy-preview-379--open-sauced.netlify.com/```, still page needs to be manually refresh to see newly added item.

**Comment by bdougie** [comment]
WIP commit pushed. Next step is to implement [reducer pattern](https://www.robinwieruch.de/react-usereducer-hook) and test

**Comment by NileshPatel17** [comment]
not sure if caching is working. is it because reducer pattern is not implemented yet.
Steps followed:-
1. goto https://deploy-preview-379--open-sauced.netlify.com/repos/nleshpatel17/a-11/8
2. then click on home icon, and take a look network tab.  You will see it makes api call.

**Comment by bdougie** [comment]
Correct @NileshPatel17, I am still working through the implementation. I got busy with work but looking at this again.

**Comment by bdougie** [comment]
Pushed an update that leverages the hook, it is not using the cache yet.

I need to figure a pattern to set the cache on the initial fetch. At the moment it will only populate the cache when the setState is used, which is also not set up.

File: src/components/RepositoryGoals.js [hunk 1]
```diff
@@ -8,24 +8,29 @@ import AddRepoForm from "../components/AddRepoForm";
...
 function RepositoryGoals() {
-  const [repository, setRepository] = useState({});
-  const [loading, setLoading] = useState(true);
   const {goalsId, setGoalsId} = useContext(LocaleContext);
+  const [state, dispatch] = usePersistentStateReducer("goalsState", goalsReducer);
+
+  const {repository} = goalsReducer(state, {type: "GET"});
+
+  useEffect(() => {
+    repository !== undefined && setGoalsId(repository.id);
```
Comment (mxschmitt) [inline src/components/RepositoryGoals.js]: not very readable, I prefer normal if.

Comment (bdougie) [reply src/components/RepositoryGoals.js]: Good feedback. This is a habit I picked up from trying to conditionally load components, but for the sake of readability I will switch to an IF

**Comment by bdougie** [comment]
Closes #348

**Pull request merged**